### DISCUSSION TITLE
Is `getServerSideProps` triggered when navigating on the client-side?

### DISCUSSION THREAD

**Discussion by italodeandra** [vercel/next.js#30049] [Help] [answered]
Hey team, if I preload a data using `getServerSideProps` does it mean that I don't need to request it again from the client on mount? If I navigate from the app to a page that has data being preloaded on `getServerSideProps` it will still get preloaded? Or as soon as the client loads the app, the `getServerSideProps` will never be called again until he opens another tab or refreshes the current page, is that right?

**Comment by jamesmosier** [comment, marked as answer]
`getServerSideProps` runs once per page and returns the data as `props` to your component. The [docs](https://nextjs.org/docs/basic-features/data-fetching#only-runs-on-server-side) mention this as well, but this is how it works:

1. On first page load (e.g. full page load or full page refresh), `getServerSideProps` runs on the server and returns the data as `props` to your page component. 
2. If you navigate to the page using the `<Link>` component (`next/link`), then Next calls an API endpoint which executes your `getServerSideProps` and returns that as JSON to your page and renders your component with those props.

**Reply by italodeandra** [reply]
Good. So does that means that I don't need to create an API to request the same thing on the client, right?

For example, I see lots of places and libraries (including SWR) telling you to use the idea of hydration to get the data from de backend with gSSP and then when the client loads, you request again the same thing. This doesn't makes sense to me. Why would I make the same request twice in both sides (back and front-end).

I understand that some people mentions that it's for updating the data you got from the backend, but why? You just requested it.

**Reply by jamesmosier** [reply]
> I don't need to create an API to request the same thing on the client, right?

That is correct

> Why would I make the same request twice in both sides (back and front-end).

It really just depends on your use-case. For most scenarios, you don't need to do this technique. Their [docs](https://swr.vercel.app/docs/with-nextjs) mention it, but the idea is when using static generation (`getStaticProps`), your page gets its data when you build your application but when you finally serve that page to the user that data might be outdated. Therefore using `swr` to re-fetch that data ensures you have the latest data for your user. 

Doing that for `getServerSideProps` wouldn't make sense, because like you mentioned you just fetched the latest data on the server and sent it to the client. But where something like the technique mentioned above might be helpful is if you wanted to fetch the data on the server (maybe for SEO reasons) and then after the page renders you want to have `swr` there to refetch the data on an interval or when a user leaves that browser tab and navigates back to it you would then refetch the data then.
